Microsoft/AdaptiveCards
# Detect collisions and generate parse errors when they are encountered

Fix for #1852.

* source/shared/**
  * Introduce new `ParseContext.cpp` and `ParseContext.h` 
    * `ParseContext` encapsulates objects that need to be passed around at parse time
      * Add new `std::unordered_set<std::string>` to track ids
      * (previously: `ElementParserRegistration`, `ActionParserRegistration`, and `std::vector<AdaptiveParseWarning>`)
  * Various wiring to replace passing the old context items with `ParseContext`
  * New unit tests for single and nested duplicate id scenarios
  * Fixes to refrain from marking default dtors as virtual (cppcore)
  * Some `#include` cleanup
  * Reorder STL includes in `pch.h` for sanity's sake

* source/uwp/**
  * Update expected results for new failing tests
  * Add new files to shared model projection
  * Updates for new `Deserialize` signature

* samples/tests
  * tests for duplicate ids in single and nested scenarios

* source/android/**
  * Fixes to consume new header/object
  * SWIG'd changes

* source/dotnet/Library/adaptivecards/**
  * Introduce new `HashSet<string>` to track element ids
  * Add id tracking during [de]serialization (throw on collision)
  * Introduce a way to indicate when a card begins/ends parsing (needed to reset our id tracking member)
  * Account for expected failures for new tests

* source/ios/**
  * Publish/consume new cpp/h
